asserts/database.go

```diff
@@ -176,3 +223,110 @@ func (db *Database) Check(assert Assertion) error {
 	}
 	return fmt.Errorf("failed signature verification: %v", lastErr)
 }
+
+func (db *Database) checkAssertionRevisionAndClearLatest(indexPath string, newRev int) error {
+	curRevStr, err := db.readlinkEntry(assertionsRoot, indexPath, "latest")
+	if err != nil {
+		return fmt.Errorf("broken assertion storage, reading 'latest' revision symlink: %v", err)
+	}
+	curRev, err := strconv.Atoi(curRevStr)
+	if err != nil {
+		return fmt.Errorf("broken assertion storage, extracting revision from 'latest' revision symlink: %v", err)
+	}
+	if curRev >= newRev {
+		return ErrNotSuperseding
+	}
+	err = db.removeEntry(assertionsRoot, indexPath, "latest")
```

Comment:
@niemeyer 'latest' is an optimisation, we can always recover by finding the top revision in the dir if it's not there, I didn't go there yet because as I said in my intro to the branch it wasn't clear to me if keeping old revision is something we want actually?